gnd/src/main/java/com/google/android/gnd/repository/ObservationRepository.java

```diff
@@ -0,0 +1,138 @@
+
+import android.util.Log;
+import com.google.android.gnd.model.feature.Feature;
+import com.google.android.gnd.model.observation.Observation;
+import com.google.android.gnd.model.observation.ObservationMutation;
+import com.google.android.gnd.persistence.local.LocalDataStore;
+import com.google.android.gnd.persistence.remote.RemoteDataStore;
+import com.google.android.gnd.persistence.remote.firestore.DocumentNotFoundException;
+import com.google.android.gnd.persistence.sync.DataSyncWorkManager;
+import com.google.android.gnd.persistence.uuid.OfflineUuidGenerator;
+import com.google.common.collect.ImmutableList;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Coordinates persistence and retrieval of {@link Observation} instances from remote, local, and in
+ * memory data stores. For more details on this pattern and overall architecture, see
+ * https://developer.android.com/jetpack/docs/guide.
+ */
+@Singleton
+public class ObservationRepository {
+  private static final String TAG = ObservationRepository.class.getSimpleName();
+  private static final long LOAD_REMOTE_RECORDS_TIMEOUT_SECS = 5;
+
+  private final LocalDataStore localDataStore;
+  private final RemoteDataStore remoteDataStore;
+  private final FeatureRepository featureRepository;
+  private final DataSyncWorkManager dataSyncWorkManager;
+  private final OfflineUuidGenerator uuidGenerator;
+
+  @Inject
+  public ObservationRepository(
+      LocalDataStore localDataStore,
+      RemoteDataStore remoteDataStore,
+      FeatureRepository featureRepository,
+      DataSyncWorkManager dataSyncWorkManager,
+      OfflineUuidGenerator uuidGenerator) {
+
+    this.localDataStore = localDataStore;
+    this.remoteDataStore = remoteDataStore;
+    this.featureRepository = featureRepository;
+    this.dataSyncWorkManager = dataSyncWorkManager;
+    this.uuidGenerator = uuidGenerator;
+  }
+
+  /**
+   * Retrieves the records or the specified project, feature, and form.
+   *
+   * <ol>
+   *   <li>Attempt to sync remote observation changes to the local data store. If network is not
+   *       available or operation times out, this step is skipped.
+   *   <li>Relevant records are returned directly from the local data store.
+   * </ol>
+   */
+  public Single<ImmutableList<Observation>> getRecords(
+      String projectId, String featureId, String formId) {
+    // TODO: Only fetch first n fields.
+    // TODO(#127): Decouple feature from observation so that we don't need to fetch observation
+    // here.
+    return featureRepository
+        .getFeature(projectId, featureId)
+        .switchIfEmpty(Single.error(new DocumentNotFoundException()))
+        .flatMap(feature -> getRecords(feature, formId));
+  }
+
+  private Single<ImmutableList<Observation>> getRecords(Feature feature, String formId) {
```

Comment:
getObservations?

Reply:
Rename of existing methods is taking place in #282.

Reply:
thanks!